lib/Doctrine/ORM/Tools/AttachEntityListenersListener.php

```diff
@@ -56,7 +56,5 @@ public function loadClassMetadata(LoadClassMetadataEventArgs $event)
                 $metadata->addEntityListener($listener['event'], $listener['class'], $listener['method']);
             }
         }
-
-        unset($this->entityListeners[$metadata->name]);
```

Comment:
Looking at this listener, what it does is attach entity listeners as soon as class metadata is loaded. After that, it considers its job done forever, hence the unset (to save memory I guess).